src/web_interface/components/database_routes.py

```diff
@@ -142,15 +141,27 @@ def _search_database(self, query, skip=0, limit=0, only_firmwares=False, inverte
 
     def _build_search_query(self):
         query = {}
-        if request.form['device_class_dropdown']:
-            query.update({'device_class': request.form['device_class_dropdown']})
-        for item in ['file_name', 'vendor', 'device_name', 'version', 'release_date']:
+        for item in ['device_class', 'vendor']:
+            if item in request.form and request.form[item]:
+                self._add_multiple_choice(query, item)
+        for item in ['file_name', 'device_name', 'version', 'release_date']:
             if request.form[item]:
                 query.update({item: {'$options': 'si', '$regex': request.form[item]}})
         if request.form['hash_value']:
             self._add_hash_query_to_query(query, request.form['hash_value'])
+        if 'tags' in request.form and request.form['tags']:
+            self._add_tag_query(query)
         return json.dumps(query)
 
+    @staticmethod
+    def _add_multiple_choice(query, key):
+        query[key] = {"$in": [name for name in dict(request.form.lists())[key]]}
+
+    @staticmethod
+    def _add_tag_query(query):
+        tag_query = [{f'tags.{tag}': 'secondary' for tag in dict(request.form.lists())['tags']}]
+        query.setdefault('$or', []).extend(tag_query)
```

Comment:
I only just noticed that this will also not do exactly what we want: this matches only firmwares that have all selected tags, but it would intuitively make more sense if it matched any firmware with at least one of the tags. Accomplishing this isn't very complicated, though. This should do it:
```suggestion
        tag_queries = [{f'tags.{tag}': 'secondary'} for tag in dict(request.form.lists())['tags']]
        query.setdefault('$or', []).extend(tag_queries)
```